File: 3940-limit-occurrences-in-sorted-array/3940-limit-occurrences-in-sorted-array.cpp
class Solution {
public:
    vector<int> limitOccurrences(vector<int>& a, int k) {
        vector <int> ans;
        int count = 1;
        ans.push_back(a[0]);
        for(int i =1; i<a.size(); i++){
            if(a[i] == a[i-1]){
                count++;
            }
            else count = 1;
            if(count <= k){
                ans.push_back(a[i]);
            }
        }return ans;
    }
};

File: 3940-limit-occurrences-in-sorted-array/3940-limit-occurrences-in-sorted-array.java
class Solution {
    public int[] limitOccurrences(int[] nums, int k) {
        int ct=1;int curr=0;
        for(int i=0;i<nums.length;i++){
            if(i>0 && nums[i]==nums[i-1]){
                ct++;
            }if(i>0 && nums[i]!=nums[i-1]){
                ct=1;
            }
            if(ct<=k){
                nums[curr]=nums[i];
                curr++;
            }
        }return Arrays.copyOf(nums,curr);
    }
}